src/components/NetworkSelector/useNetworkName.ts

```diff
@@ -1,22 +1,24 @@
 import useSWR from 'swr'
-import LotusRPCEngine from '@glif/filecoin-rpc-client'
+import LotusRpcEngine from '@glif/filecoin-rpc-client'
+import { useEnvironment } from '../../services/EnvironmentProvider'
 
-const fetcher = async (lotusApiAddr: string) => {
-  const lCli = new LotusRPCEngine({
-    apiAddress: lotusApiAddr
-  })
-  const network = (await lCli.request('StateNetworkName')) as string
-  if (network) return network
-
-  throw new Error('Not connected to a network')
+interface UseNetworkNameResult {
+  networkName: string
+  loading: boolean
+  error: Error | null
 }
 
-export const useNetworkName = (lotusApiAddr: string) => {
-  const {
-    data: networkName = '',
-    error = null,
-    isValidating
-  } = useSWR(lotusApiAddr, fetcher)
+const fetcher = async (
+  lotusApi: LotusRpcEngine,
+  lotusMethod: string
+): Promise<string> => await lotusApi.request<string>(lotusMethod)
+
+export const useNetworkName = (): UseNetworkNameResult => {
+  const { lotusApi } = useEnvironment()
+  const { data, error } = useSWR<string, Error>(
+    [lotusApi, 'StateNetworkName'],
+    fetcher
+  )
 
-  return { networkName, error, loading: isValidating }
+  return { networkName: data ?? '', loading: !data && !error, error }
```

Comment:
`isValidating` is also true when there is a `refreshInterval`. According to the docs it's best to check for the initial loading state with `!data && !error`